server/container_create.go

```diff
@@ -124,29 +124,111 @@ func addImageVolumes(rootfs string, s *Server, containerInfo *storage.ContainerI
 	return nil
 }
 
+// resolveSymbolicLink resolves a possbile symlink path. If the path is a symlink, returns resolved
+// path; if not, returns the original path.
+func resolveSymbolicLink(path string) (string, error) {
```

Comment:
fileutils.ReadSymlinkedDirectory(path) does something like this also.

Reply:
Do you want me to use that? 
```
244 // ReadSymlinkedDirectory returns the target directory of a symlink.
245 // The target of the symbolic link may not be a file.
246 func ReadSymlinkedDirectory(path string) (string, error) {
247     var realPath string
248     var err error
249     if realPath, err = filepath.Abs(path); err != nil {
250         return "", fmt.Errorf("unable to get absolute path for %s: %s", path, err)
251     }
252     if realPath, err = filepath.EvalSymlinks(realPath); err != nil {                                                       253         return "", fmt.Errorf("failed to canonicalise path for %s: %s", path, err)
254     }
255     realPathInfo, err := os.Stat(realPath)
256     if err != nil {
257         return "", fmt.Errorf("failed to stat target '%s' of '%s': %s", realPath, path, err)
258     }
259     if !realPathInfo.Mode().IsDir() {
260         return "", fmt.Errorf("canonical path points to a file '%s'", realPath)
261     }
262     return realPath, nil
263 }
```